File: E_Partial_sum_computation.py
x = int(input())

for i in range(x):
    n = int(input())
    arr = list(map(int,input().split())) 
    arr.sort()
    cur_sum =0 
    ps = [0]*(n)
    flag = False
    if arr[0] != 1:
        print("NO")
    else:
        for i in range(n):
            cur_sum += arr[0]
            ps[i] += cur_sum


        for i in range(1,n):
            if arr[i] < ps[i-1]:
                print("No")
                break
        else:
            print("YES")

    

        

File: E_Partial_sum_computation.c
import math
 
 
def solve():
    n, m, q = map(int, input().split())
    arr = []
    dist = dict()
    S = []
    for _ in range(n):
        arr.append(input())
 
    _ = int(input())
    s = input().strip()
 
    for i in range(len(arr)):
        for j in range(len(arr[i])):
            if arr[i][j] == 'S':
                S.append((i, j))
            else:
                dist[arr[i][j]] = 0
 
    if len(S) > 0:
        for i in range(n):
            for j in range(m):
                if arr[i][j] != 'S':
                    # sqrt(pow(x1-x2, 2) + pow(y1-y2, 2))
                    mn = min(math.sqrt((pow(i - shift[0], 2) + pow(j - shift[1], 2))) for shift in S)
                    cap = arr[i][j].upper()
                    if mn <= q:
                        dist[cap] = 0
                    elif cap not in dist:
                        dist[cap] = 1
    res = 0
    for el in s:
        if el not in dist:
            print(-1)
            return
        else:
            res += dist[el]
    print(res)
 
    return
 
 
def main():
    # test = int(input())
    # for t in range(test):
    #     solve()
 
    solve()
 